### ISSUE TITLE
Righthand menu should always be visible

### ISSUE THREAD

**Issue by JirkaDellOro** [TypeStrong/typedoc#985]
Hello,

I'm starting to use typdoc. I think the resulting documentation could be improved easily and a lot, if e.g. for a class, the righthand menu was always visible. At this point of time, when I click on a method, the whole content scrolls up and the menu vanishes. It would be much better, if only the left side scrolls. Then I can always see the overview on the right and the details on the left. I was able to approximate this by changing line 748 of the default-css to
```
  .menu-sticky-wrap { position: sticky; }
```
but there are sure unwanted side-effects to this.

Is there an appropriate way right now to achieve what I'm looking for?

Thanks for the good work!

**Comment by aciccarello** [comment]
Thanks for logging this. I actually realize this this week looking at the themes. If you see a way to fix the sticky sidebar please pass along your findings. I'd love to get this fixed, particularly if we could drop the JavaScript for the sticky menu we currently have.

**Comment by JirkaDellOro** [comment]
Ah, ok, so what we see at the moment is not the intended behaviour. Actually, "sticky" suggests it. 

I do have the feeling, that the whole css is much more complex than necessary. However, I'm just starting to use TypeDoc and I'm not a web-specialist. It would be great if some CSS-wiz could take a look at it.

**Comment by BennyHinrichs** [comment]
So there are two possibilities. One is to get rid of the JS entirely and put `position: sticky; top: 1rem` on `.menu-sticky-wrap`. That will make the whole thing scroll.

However, it looks like you don't want to include the globals? In that case, you'll have to do the top calculation in an initial load. I would do it as follows, but there's likely some different way that would play nicer with the whole framework you have set up.

```javascript
const primaryNavHeight = document.querySelector('.tsd-navigation.primary').clientHeight;
document.querySelector('.menu-sticky--wrap').style.top = `calc(1rem - ${primaryNavHeight})`;
```

An element is sticky relative to its parent. The parent of `.tsd-navigation.secondary` is `.menu-sticky-wrap`, and `.tsd-navigation.secondary` is at the bottom. If you just put sticky on the secondary nav, it will be sticky, but you'll never see the effects because it's at the bottom of its parent. `.tsd-navigation.primary` is at the top of its parent, so we get the desired effect.

**Comment by BennyHinrichs** [comment]
So this is what I did.
`MenuSticky.ts`: Got rid of entirely

`MenuHighlight.ts`: Added the following lines to the constructor along with their private declarations above
```javascript
this.$primary = this.$el.find('.tsd-navigation.primary');
this.stickyTop = this.$primary.height() || 0;
this.$el.css({'top': `calc(-${this.stickyTop}px - 40px)`});
```
I combined Highlight and Sticky because there's a bit of code in `Application.ts` that only lets you have one `instance` per element. This could potentially be renamed to `Menu.ts` which has the Sticky and Highlight classes inside and calls those, like how `Filter.ts` does.

`_navigation.sass`: I changed the "Sticky menu startup" portion to
```css
.menu-sticky-wrap
    +size-md-lg
        position: sticky
        top: 1rem
```

**Comment by BennyHinrichs** [comment]
Wow, I'm dumb. I just realized you could fix the whole thing with three CSS attributes.

```css
.row
    display: flex;

.tsd-navigation
    &.secondary
        position: sticky
        top: .5rem
```

Add that to the CSS then delete `MenuSticky.ts` entirely and it fixes the problem! Also have to delete the reference to `MenuSticky.ts` in `~Bootstrap.ts`.

I will go about making a pull request for this in the `typedoc-default-themes` repo. I've never done one on GitHub before so with me luck!

**Comment by JirkaDellOro** [comment]
Sounds great!

**Comment by Gerrit0** [comment]
Fixed by v0.6.1 of the themes :)
